src/Xamarin.Android.Build.Tasks/Tasks/GetJavaPlatformJar.cs

```diff
@@ -58,22 +65,24 @@ public override bool RunTask ()
 								platform = target_sdk.Value;
 
 							var min_sdk = uses_sdk.Attribute (androidNs + "minSdkVersion");
-							if (min_sdk != null && (!int.TryParse (min_sdk.Value, out int minSdkVersion) || minSdkVersion < XABuildConfig.NDKMinimumApiAvailable)) {
-								Log.LogWarningForXmlNode (
-										code:             "XA4216",
-										file:             AndroidManifest,
-										node:             min_sdk,
-										message:          Properties.Resources.XA4216_MinSdkVersion,
-										messageArgs:      new object [] { min_sdk?.Value, XABuildConfig.NDKMinimumApiAvailable }
-								);
+							if (min_sdk != null) {
+								var failedToParseMinSdk = !int.TryParse (min_sdk.Value, out int minSdkVersion);
+
+								if (failedToParseMinSdk || minSdkVersion < XABuildConfig.AndroidMinimumDotNetApiLevel) {
+									Log.LogCodedError ("XA4216", Properties.Resources.XA4216_MinSdkVersion, min_sdk?.Value, XABuildConfig.AndroidMinimumDotNetApiLevel);
+								}
+
+								if (failedToParseMinSdk || minSdkVersion != supportedOsPlatformVersionAsInt) {
+									Log.LogCodedError ("XA1036", Properties.Resources.XA1036, min_sdk?.Value, SupportedOSPlatformVersion);
```

Comment:
Is it possible to provide a filename for the error message?

Reply:
I don't know of an easy way to report what file is setting `SupportedOSPlatformVersion` (it will likely be set in the project file, but could be in a directory.build.props or any other import).